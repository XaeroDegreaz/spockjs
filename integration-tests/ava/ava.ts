import { resolve } from 'path';

import { modulePath, runWithTypescriptJit } from '../utils';

const avaCli = resolve(modulePath('ava'), 'cli.js');

export const runAva = (cwd: string) => {
  const { status, stdout } = runWithTypescriptJit(
    [avaCli, '--serial', '--no-cache', '--verbose'],
    { cwd },
  );

  return {
    status,
    stdout: stdout
      .toString()
      .replace(/✔|√|✖|×/g, ' ') // remove OS-dependent symbols
      .replace(/[^ ]+ava\.js:/, '<FILEPATH HIDDEN>:'), // remove absolute path
  };
};
