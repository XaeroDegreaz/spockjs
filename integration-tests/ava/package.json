{
  "ava": {
    "files": ["ava.js"],
    "failWithoutAssertions": false,
    "babel": {
      "testOptions": {
        "plugins": ["../../packages/babel-plugin-spock/src/index.ts"]
      }
    }
  },
  "babel": {
    "presets": [["@babel/preset-env", { "targets": { "node": 6 } }]]
  }
}
