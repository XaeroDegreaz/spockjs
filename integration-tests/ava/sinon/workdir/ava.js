import test from 'ava';
import * as sinon from 'sinon';

test('stubbing', t => {
  const s = sinon.stub();
  stub: s() >> 42;
  t.truthy(s() === 42);
});

test('mocking', t => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();
  obj.method();

  verify: mock;
});

test('failing verification', t => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();

  verify: mock;
});
