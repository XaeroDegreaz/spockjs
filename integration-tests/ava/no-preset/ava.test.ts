import { resolve } from 'path';

import { runAva } from '../ava';

// mark implicit dependencies for jest
() => require('./workdir/ava.js') && require('./workdir/package.json');

const cwd = resolve(__dirname, 'workdir');

test('produces very verbose output by default', () => {
  const { status, stderr } = runAva(cwd);

  expect: status === 1;
  expect(stderr).toMatchSnapshot();
});
