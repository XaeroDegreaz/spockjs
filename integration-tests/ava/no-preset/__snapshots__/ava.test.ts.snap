// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`produces very verbose output by default 1`] = `
"
    successful test
    failing test Error thrown in test

  1 test failed

  failing test

  <FILEPATH HIDDEN>:12

   11:   when: x = 3;          
   12:   then: x === double(2);
   13: });                     

  Error thrown in test:

  AssertionError [ERR_ASSERTION] (AssertionError) {
    actual: false,
    code: 'ERR_ASSERTION',
    expected: true,
    generatedMessage: false,
    operator: '==',
    powerAssertContext: {
      args: [
        Object { … },
      ],
      source: {
        ast: Object { … },
        content: '_assert(x === double(2), \\"(x === double(2)) is not truthy\\")',
        filepath: 'ava.js',
        line: 12,
        tokens: Array [ … ],
        visitorKeys: Object { … },
      },
    },
    message: \`(x === double(2)) is not truthy   # ava.js:12␊
      ␊
      _assert(x === double(2), \\"(x === double(2)) is not truthy\\")␊
              | |   |                                            ␊
              | |   4                                            ␊
              3 false                                            ␊
      ␊
      [number] double(2)␊
      => 4␊
      [number] x␊
      => 3␊
      \`,
  }

"
`;
